backend/tests/unit/dal/Services/Admin/UserServiceTest.cs

```diff
@@ -0,0 +1,214 @@
+
+            // Assert
+            Assert.Equal(1, result);
+        }
+        #region Get
+        [Fact]
+        public void Get_Users_Paged()
+        {
+            // Arrange
+            var helper = new TestHelper();
+            var user = PrincipalHelper.CreateForPermission(Permissions.AdminUsers);
+            var euser = EntityHelper.CreateUser("Tester");
+            helper.CreatePimsContext(user, true).AddAndSaveChanges(euser);
+            var expectedCount = 1;
+
+            var service = helper.CreateService<UserService>(user);
+
+            // Act
+            var result = service.Get(1, 1);
+
+            // Assert
+            Assert.NotNull(result);
+            Assert.IsAssignableFrom<Paged<Entity.User>>(result);
+            Assert.Equal(expectedCount, result.Items.Count());
+        }
+
+        [Fact]
+        public void Get_Users_NotAuthorized()
+        {
+            // Arrange
+            var helper = new TestHelper();
+            var user = PrincipalHelper.CreateForPermission();
+
+            var service = helper.CreateService<UserService>(user);
+
+            // Act
+            // Assert
+            Assert.Throws<NotAuthorizedException>(() =>
+                service.Get());
+        }
+
+        [Theory]
+        [MemberData(nameof(UserFilterData))]
+        public void Get_Users_Filter(UserFilter filter, int expectedCount)
+        {
+            // Arrange
+            var helper = new TestHelper();
+            var user = PrincipalHelper.CreateForPermission(Permissions.AdminUsers);
+            var euser = EntityHelper.CreateUser("Tester");
+            euser.FirstName = "Test";
+            euser.LastName = "McTest";
+            euser.DisplayName = "ttester";
+            euser.Email = "test@test.com";
+            euser.IsDisabled = false;
+            helper.CreatePimsContext(user, true).AddAndSaveChanges(euser);
+
+            var service = helper.CreateService<UserService>(user);
+
+            // Act
+            var result = service.Get(filter);
+
+            // Assert
+            Assert.NotNull(result);
+            Assert.IsAssignableFrom<Paged<Entity.User>>(result);
+            Assert.Equal(expectedCount, result.Items.Count());
+        }
+
+        [Fact]
+        public void Get_User_ById()
+        {
+            // Arrange
+            var helper = new TestHelper();
+            var user = PrincipalHelper.CreateForPermission(Permissions.AdminUsers);
+            var id = Guid.NewGuid();
+            var euser = EntityHelper.CreateUser(id, "ttester", "Tester", "McTest");
+            helper.CreatePimsContext(user, true).AddAndSaveChanges(euser);
+
+            var service = helper.CreateService<UserService>(user);
+
+            // Act
+            var result = service.Get(id);
+
+            // Assert
+            Assert.NotNull(result);
+            Assert.Equal(id, result.Id);
+        }
+        #endregion
+        #region Add
+        [Fact]
+        public void Add_User()
```

Comment:
unauth tests for this service?

Reply:
Yeah, a number of these tests are still in progress, was just throwing up some of the ones I had to make sure I wasn't completely off the mark. I am tracking them all with codecov!